src/main/java/com/hierynomus/smbj/smb2/messages/SMB2IoctlRequest.java

```diff
@@ -59,15 +57,24 @@ protected void writeTo(SMBBuffer smbBuffer) {
         fileId.write(smbBuffer);  // FileId (16 bytes)
 
         int offset = SMB2Header.STRUCTURE_SIZE + 56;
-        smbBuffer.putUInt32(offset); // InputOffset (4 bytes)
-        smbBuffer.putUInt32(inputData.length); // InputCount (4 bytes)
-        smbBuffer.putUInt32(MAX_OUTPUT_BUFFER_LENGTH); // MaxInputResponse (4 bytes)
+        if (inputData.length > 0) {
+            smbBuffer.putUInt32(offset); // InputOffset (4 bytes)
+            smbBuffer.putUInt32(inputData.length); // InputCount (4 bytes)
+        } else {
+            smbBuffer.putUInt32(0); // InputOffset (4 bytes)
```

Comment:
Ok, just read up on this field. The question is, how to interpret this:

```
The offset, in bytes, from the beginning of the SMB2 header to the input data buffer. If no input data is required for the FSCTL/IOCTL command being issued, the client SHOULD set this value to 0.
```

Do we know when no input data is required, do we need to add some check to verify the conditions when inputdata is required?